urllib3/backends/trio_backend.py

```diff
@@ -50,18 +55,18 @@ def getpeercert(self, binary=False):
 
         async def receiver():
             while True:
-                incoming = await stream.receive_some(BUFSIZE)
+                incoming = await self._stream.receive_some(BUFSIZE)
                 consume_bytes(incoming)
 
         try:
             async with trio.open_nursery() as nursery:
-                nursery.spawn(sender)
-                nursery.spawn(receiver)
+                nursery.start_soon(sender)
+                nursery.start_soon(receiver)
         except LoopAbort:
             pass
 
-    def forceful_close(self):
-        self._stream.forceful_close()
+    async def forceful_close(self):
+        await trio.aclose_forcefully(self._stream)
```

Comment:
Ugh, forceful close raises such tricky questions for API design.

I started to write a long thing here about why `close` should maybe be synchronous after all. Trio's abstract stream interface makes it async because it's, y'know, abstract, and some streams need it to be async. But for the particular concrete streams that urllib3 uses, it's always synchronous. So maybe, I was thinking, it should be synchronous here too, and we'd make the trio backend jump through the necessary hoops to make that work.

Then l looked at twisted and asyncio and realized that they also make closing a socket an async operation for weird API reasons, so fine, let's just make it async :-). It's an issue we might want to revisit in the future though as this develops. (I'm also inclined to say that if we're going to make closing async, we should rename it `aclose`, which is the convention that trio uses; I stole it from the interpreter's async generator API. But that's also something we can worry about later.)

Reply:
Thanks for the explanation. So `forceful_close` should become `aclose_forcefully`?

Reply:
Oh, I guess it could. That doesn't matter too much though, because the backend API is internal anyway. The more important question is whether we want to rename `Response.close`, `PoolManager.close`, etc.

Reply:
Okay, I'll prepare another PR for this change.

Reply:
Just opened #4.